redbot/cogs/warnings/warnings.py

```diff
@@ -301,19 +342,39 @@ def __init__(self, bot: Red):
 
         await warning_points_add_check(self.config, ctx, user, current_point_count)
         try:
+            dm = await self.config.guild(ctx.guild).toggle_dm()
+            if dm:
+                em = discord.Embed(
+                    title=_("Warning from {user}").format(user=ctx.author),
+                    description=reason_type["description"],
+                )
+                em.add_field(name=_("Points"), value=str(reason_type["points"]))
+                await user.send(
+                    _("You have received a warning in {guild_name}.").format(
+                        guild_name=ctx.guild.name
+                    ),
+                    embed=em,
+                )
+        except discord.HTTPException:
+            await ctx.send(
+                "The warning couldn't be sent to the user because the user has DM's from server members turned off."
+            )
+        await ctx.send(_("User {user} has been warned in their DM's.").format(user=user))
+        toggle_channel = await self.config.guild(ctx.guild).toggle_channel()
+        if toggle_channel:
             em = discord.Embed(
                 title=_("Warning from {user}").format(user=ctx.author),
                 description=reason_type["description"],
             )
             em.add_field(name=_("Points"), value=str(reason_type["points"]))
-            await user.send(
-                _("You have received a warning in {guild_name}.").format(
-                    guild_name=ctx.guild.name
-                ),
-                embed=em,
-            )
-        except discord.HTTPException:
-            pass
+            warn_channel = await self.config.guild(ctx.guild).warn_channel()
+            if warn_channel is None:
+                channel = ctx.channel
+            else:
+                channel = warn_channel
+                await channel.send(
```

Comment:
I think this could work nicely if you would unindent this `channel.send()` once, so that it gets sent, when warn channel is not set too (see my proposed addition in comment to `[p]warningset warnchannel` for additional context)